app/src/main/java/ca/marshallasch/veil/FragmentLanding.java

```diff
@@ -0,0 +1,61 @@
+package ca.marshallasch.veil;
+
+import android.app.Fragment;
+import android.app.FragmentManager;
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.support.design.button.MaterialButton;
+import android.support.v4.app.FragmentTransaction;
+import android.util.Log;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+
+public class FragmentLanding extends Fragment {
+
+    public FragmentLanding() {
+        // Required empty public constructor
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        ((MainActivity) getActivity()).getSupportActionBar().hide();
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState)
+    {
+        View view = inflater.inflate(R.layout.fragment_landing, container, false);
+
+        //login button and event listener
+        MaterialButton loginBtn = view.findViewById(R.id.login_btn);
+        loginBtn.setOnClickListener(view1 -> {
+            Log.i("Fragment Landing", "login button pressed");
+            ((MainActivity) getActivity()).navigateTo(new SignUp(), falsea);
+            ((MainActivity) getActivity()).getSupportActionBar().show();
```

Comment:
You have duplicated code here and on line 47, and 58. 
you can change showing /hiding the action bar to the [fragment life cycle](https://developer.android.com/guide/components/fragments#Creating) in the `onCreateView()` and the `onDestroyView()` call backs instead.

Reply:
resolved